Add HostSNIRegexp rule matcher for TCP (#8849)
traefik/traefik: docs/content/routing/routers/index.md

The table below lists all the available matchers:

| Rule                                                                      | Description                                                                                               |
|---------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------|
| ```HostSNI(`domain-1`, ...)```                                            | Check if the Server Name Indication corresponds to the given `domains`.                                   |

> | ```HostSNIRegexp(`example.com`, `{subdomain:[a-z]+}.example.com`, ...)``` | Check if the Server Name Indication is matching the given regular expressions. See "Regexp Syntax" below. |

Reviewer: ```suggestion
| ```HostSNIRegexp(`example.com`, `{subdomain:[a-z]+}.example.com`, ...)``` | Check if the Server Name Indication matches the given regular expressions. See "Regexp Syntax" below. |
```

| ```ClientIP(`10.0.0.0/16`, `::1`)```                                      | Check if the request client IP is one of the given IP/CIDR. It accepts IPv4, IPv6 and CIDR formats.       |

!!! important "Non-ASCII Domain Names"

>     Non-ASCII characters are not supported in the `HostSNI` and `HostSNIRegexp` expressions, and by doing so the associated TCP router will be invalid.

Reviewer: ```suggestion
    Non-ASCII characters are not supported in the `HostSNI` and `HostSNIRegexp` expressions, and so using them would invalidate the associated TCP router.
```
